packages/connection-form/src/components/advanced-options-tabs/authentication-tab/authentication-gssapi.tsx

```diff
@@ -99,9 +143,45 @@ function AuthenticationGSSAPI({
               value: value,
             });
           }}
-          label={GSSAPI_SERVICE_REALM_LABEL}
+          data-testid="gssapi-service-realm-input"
+          label="Service Realm"
           value={serviceRealm || ''}
+          optional
+        />
+      </FormFieldContainer>
+      <FormFieldContainer>
+        <Checkbox
+          data-testid="gssapi-password-checkbox"
+          checked={showPassword}
+          label="Provide password directly"
+          onChange={({ target: { checked } }) => {
+            if (!checked) {
+              updateConnectionFormField({
+                type: 'update-password',
+                password: '',
+              });
+            }
+
+            setShowPassword(checked);
+          }}
         />
+        {showPassword && (
+          <TextInput
+            onChange={({
+              target: { value },
+            }: React.ChangeEvent<HTMLInputElement>) => {
+              updateConnectionFormField({
+                type: 'update-password',
+                password: value,
+              });
+            }}
+            data-testid="gssapi-password-input"
+            label="Password"
+            value={password}
+            type="password"
+            optional
+          />
+        )}
```

Comment:
This should probably wrapped in its own `FormFieldContainer`? The paddings of this field look a bit weird when it's visible right now